addons/controllers/webhooks/manifests.yaml

```diff
@@ -0,0 +1,95 @@
+---
```

Comment:
Could we define and create these configurations with GO code instead of defining them in yaml file? I feel in this way it would be easier to change and maintain.

Reply:
These configurations are auto-generated by using kubebuilder. They are generated from the markers that are in the scaffolded code. If we make all these in GO code, then I think it is harder to maintain. Thinking aloud here - say we need to add webhooks for another component… we will not have to do anything at all except adding 2 lines to call the function SetupWebhookWithManager for that new component, and of course implement the logic in validating and mutating webhooks. Everything else will be autogenerated. It felt much easier doing this way, both in terms of how much changes we need to do and also to maintain. But if the team decides to go with GO code instead, I can change accordingly.

Reply:
We probably want to move to using the yml files and just using a CrateResources(). 
in other words. Generate the yaml files which we will eventually pacakge along with the addons package, and apply them. that way we get to test all at once.

Reply:
I think part of this pr's purpose IS to create the yaml files.

Reply:
^^ I take that back. I guess this pr is not about manifests.yamls. ?

Reply:
As mentioned above, the manifests.yaml is automatically generated using the markers. So, it is just a part of this change.